packages/scaffold-ui/src/views/w3m-networks-view/index.ts

```diff
@@ -146,48 +146,15 @@ export class W3mNetworksView extends LitElement {
     return !approvedCaipNetworkIds?.includes(network.id)
   }
 
-  private async onSwitchNetwork(network: CaipNetwork) {
-    const isCurrentNamespaceConnected = AccountController.state.caipAddress
-    const isNamespaceConnected = AccountController.getCaipAddress(network.chainNamespace)
-    const isSameNetwork = network.id === this.network?.id
-
-    const supportsAllNetworks = NetworkController.state.supportsAllNetworks
+  private onSwitchNetwork(network: CaipNetwork) {
     const routerData = RouterController.state.data
-
-    const type = StorageUtil.getConnectedConnector()
-    const authConnector = ConnectorController.getAuthConnector()
-    const isConnectedWithAuth = type === 'AUTH' && authConnector
+    const isSameNetwork = network.id === this.network?.id
 
     if (isSameNetwork) {
       return
     }
 
-    if (isNamespaceConnected) {
-      if (supportsAllNetworks || isConnectedWithAuth) {
-        RouterController.push('SwitchNetwork', { ...routerData, network })
-      } else {
-        await NetworkController.switchActiveNetwork(network)
-      }
-    } else {
-      // eslint-disable-next-line no-lonely-if
-      if (ChainController.state.noAdapters) {
-        RouterController.push('ConnectingWalletConnect')
-      } else if (isConnectedWithAuth) {
-        RouterController.push('SwitchNetwork', { ...routerData, network })
-      } else {
-        // eslint-disable-next-line no-lonely-if
-        if (isCurrentNamespaceConnected) {
-          RouterController.push('SwitchActiveChain', {
-            switchToChain: network.chainNamespace,
-            navigateTo: 'Connect',
-            navigateWithReplace: true,
-            network
-          })
-        } else {
-          NetworkController.setActiveCaipNetwork(network)
-        }
-      }
-    }
+    RouterController.push('SwitchNetwork', { ...routerData, network })
```

Comment:
What about the other cases? Like multiple adapter flows (`SwitchActiveChain`)? or `ConnectingWalletConnect` case? I don't see the other cases handled in the `SwitchNetwork` view

Reply:
I'm still in progress with that, this was one attempt to clean up the logic because I could not figure out what was intended to happen here. But I'll jam more over to get it sorted.